If the bond bridge port does not have a slave it would not be assigned a mac,
in that case the bridge mac would be all zero too. Avoid picking
a port which has all zero mac for the bridge address.

diff --git a/net/bridge/br_stp_if.c b/net/bridge/br_stp_if.c
index 4ed11ab..f440111 100644
--- a/net/bridge/br_stp_if.c
+++ b/net/bridge/br_stp_if.c
@@ -246,8 +246,9 @@ bool br_stp_recalculate_bridge_id(struct net_bridge *br)
 		return false;
 
 	list_for_each_entry(p, &br->port_list, list) {
-		if (addr == br_mac_zero ||
-		    memcmp(p->dev->dev_addr, addr, ETH_ALEN) < 0)
+		if ((addr == br_mac_zero ||
+		    memcmp(p->dev->dev_addr, addr, ETH_ALEN) < 0) &&
+			!ether_addr_equal(p->dev->dev_addr, br_mac_zero))
 			addr = p->dev->dev_addr;
 
 	}
